### PR TITLE
feat(x/ecocredit): query `Batches` ORM

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

adds batches query

ref: #696

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/query_batches.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		return nil, err
+	}
+	project, err := k.stateStore.ProjectInfoStore().GetByName(ctx, request.ProjectId)
+	if err != nil {
+		return nil, err
+	}
+	it, err := k.stateStore.BatchInfoStore().List(ctx, ecocreditv1beta1.BatchInfoProjectIdIndexKey{}.WithProjectId(project.Id), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
```
Comment (ryanchristo) [inline x/ecocredit/server/core/query_batches.go:23]: We only need the id from project, which we already have in the request. Do we need to get the project from state or can we just convert the id in the request?

File: x/ecocredit/server/core/query_batches_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	assert.ErrorContains(t, err, ormerrors.NotFound.Error())
+
+	//  base query
+	res, err := s.k.Batches(s.ctx, &v1beta1.QueryBatchesRequest{ProjectId: "P01"})
+	assert.NilError(t, err)
+	assert.Equal(t, 2, len(res.Batches))
+	assert.Equal(t, "C01-20200101-20220101-001", res.Batches[0].BatchDenom)
+
+	// paginated query
+	res, err = s.k.Batches(s.ctx, &v1beta1.QueryBatchesRequest{ProjectId: "P01", Pagination: &query.PageRequest{Limit: 1, CountTotal: true}})
```
Comment (ryanchristo) [inline x/ecocredit/server/core/query_batches_test.go:49]:
```suggestion
	res, err = s.k.Batches(s.ctx, &v1beta1.QueryBatchesRequest{
		ProjectId: "P01",
		Pagination: &query.PageRequest{Limit: 1, CountTotal: true},
	})
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

File: x/ecocredit/server/core/query_batches.go (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/server/core/query_batches.go:23]: nvm

File: x/ecocredit/server/core/query_batches.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Batches queries for all batches in the given credit class.
+func (k Keeper) Batches(ctx context.Context, request *v1beta1.QueryBatchesRequest) (*v1beta1.QueryBatchesResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+	project, err := k.stateStore.ProjectInfoStore().GetByName(ctx, request.ProjectId)
```
Comment (amaurym) [inline x/ecocredit/server/core/query_batches.go:16]: a bit confusing that the ProjectId in the request refers to the unique project name (string), and not its DB id (uint64). But i guess it's fine thanks to the different types

**Review by amaurym — APPROVED** [review]
lgtm

**Pull request merged**